client/src/components/ItemDetails/index.js

```diff
@@ -116,158 +123,112 @@ class ItemDetails extends Component {
-    } else if (name === 'brands') {
-      const value1 = JSON.parse(value);
-      this.setState({ [`selected_${name}`]: { id: value1.id, brandName: value1.name } });
-    } else if (name === 'colors') {
-      const { colors } = this.state;
-      const color = colors.filter(x => (x.name === value ? x.hex : null));
-      this.setState({ selected_hex: color[0].hex, selected_colors: color[0].name, isOpen: false });
-    } else if (name === 'itemType') {
-      const value1 = JSON.parse(value);
-      this.setState({ [`selected_${name}`]: { id: value1.id, itemType: value1.itemType } });
-    } else {
-      this.setState({ [`selected_${name}`]: value });
-    }
-  };
-
-  toggleClose = (e) => {
-    e.preventDefault();
-    this.setState({ isOpen: false });
-  };
-
-  changeSelected = (e) => {
-    e.preventDefault();
-    const { value, name, id } = e.target;
-    const selected = `selected_${this.state.selectedCat}`;
-    if (name === 'brands') {
-      const value1 = JSON.parse(value);
-      this.setState({ [selected]: { id, brandName: value1.name, name: value }, isOpen: false });
-    } else if (name === 'colors') {
-      const { colors } = this.state;
-      const color = colors.filter(x => (x.name === value ? x.hex : null));
-      this.setState({ selected_hex: color[0].hex, selected_colors: color[0].name, isOpen: false });
-    } else if (name === 'itemType') {
-      const value1 = JSON.parse(value);
-      this.setState({
-        [`selected_${name}`]: { id, itemType: value1.itemType, name: value },
-        isOpen: false,
-      });
-    } else {
-      this.setState({ [selected]: value, isOpen: false });
-    }
+    this.setState({ [`selected${name}`]: value });
   };
 
-  checkForChanges = () => {
-    const {
-      selected_brands,
-      selected_condition,
-      selected_labelSize,
-      selected_age,
-      selected_colors,
-      selected_itemType,
-      selected_price,
-      selected_currency,
-      selected_details,
-      selected_sizeCategory,
-      selected_patterns,
-      itemDetails,
-    } = this.state;
-
-    const itemDetailsArray = [];
-    const updateItemDetails = [];
-
-    const keys = Object.keys(itemDetails);
-
-    keys.map((key) => {
-      itemDetailsArray.push(itemDetails[key]);
-    });
-
-    updateItemDetails.push(
-      itemDetails.itemId,
-      selected_itemType.itemType,
-      selected_itemType.id,
-      selected_labelSize,
-      itemDetails.url,
-      itemDetails.name,
-      selected_price,
-      selected_colors,
-      selected_brands.brandName,
-      selected_condition,
-      selected_age,
-      selected_details,
-      selected_brands.id,
-      itemDetails.colors,
-      selected_sizeCategory,
-      selected_patterns,
-    );
-
-    let flag = false;
-    itemDetailsArray.map((object, i) => {
-      if (object !== updateItemDetails[i]) {
-        flag = true;
-      }
-    });
-
-    return flag;
-  };
+  // checkForChanges = () => {
```

Comment:
as this is removed it will call /edit-item endpoint whether there's a change or not